src/Uno.UI/Media/NativeRenderTransformAdapter.iOSmacOS.cs

```diff
@@ -64,6 +67,17 @@ partial class NativeRenderTransformAdapter
 			}
 		}
 
+		private void InitializeOrigin()
+		{
+			var oldFrame = Owner.Layer.Frame;
+
+			Owner.Layer.AnchorPoint = CurrentOrigin;
+
+			// Restore the old frame to correct the offset potentially introduced by changing AnchorPoint. This is safe to do since we know
+			// that the transform is currently identity.
+			Owner.Layer.Frame = oldFrame;
```

Comment:
You're not altering it...  why do you need such restore?

Reply:
Setting AnchorPoint alters it. Actually Frame is a 'composed property' that depends on Bounds, AnchorPoint, Position.